File: 1.x/demos/iOS/Example-iOS-ObjC/Example-iOS-ObjC/CancelTransactionController.m
//
//  CancelTransactionController.m
//  Example-iOS-ObjC
//
//  Created by Hildequias Junior on 8/8/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

#import "CancelTransactionController.h"
#import "PlugPag.h"
#import "UIUtils.h"

#define MSG_CANCEL_SUCESS       @"Transação cancelada com sucesso!"
#define MSG_ERROR_DEVICE_NULL   @"Não foi possível efetuar uma conexão com o dispositivo pareado: %i"
#define MSG_ERROR_GENERIC       @"Não foi possível efetuar a operação"

@interface CancelTransactionController ()

@end

@implementation CancelTransactionController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    [[PlugPag sharedInstance] SetVersionName:@"UnitTest" withVersion:@"V001"];
}

- (void) viewDidAppear:(BOOL)animated {
    
    [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)cancelTransaction:(UIButton *)sender {
    
    [sender setEnabled:NO];
    [UIUtils showProgress];
    
    int ret = [[PlugPag sharedInstance] InitBTConnection];
    
    switch (ret) {
        case RET_OK:
            [self cancelTransaction];
            break;
            
        case ERROR_DEVICE_NULL:
            _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_DEVICE_NULL];
            break;
            
        default:
            _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_GENERIC];
            break;
    }
    
    [UIUtils hideProgress];
    [sender setEnabled:YES];
}

- (void) cancelTransaction {
    
    int ret = [[PlugPag sharedInstance] CancelTransaction];
    
    if (ret == RET_OK) {
        _txtResult.text = MSG_CANCEL_SUCESS;
    }else{
        _txtResult.attributedText = [UIUtils formatTextError:ret withMessage: MSG_ERROR_GENERIC];
    }
}

@end

File: demos/iOS/Example-iOS-Swift/Example-iOS-Swift/CancelTransactionController.swift
//
//  CancelTransactionController.swift
//  Example-iOS-Swift
//
//  Created by Hildequias Junior on 8/23/17.
//  Copyright © 2017 UOL Pagseguro. All rights reserved.
//

import UIKit

class CancelTransactionController: UIViewController {

    @IBOutlet weak var txtResult: UITextView?
    @IBOutlet weak var btCancel: UIButton?
    
    let MSG_CANCEL_SUCESS       = "Transação cancelada com sucesso!"
    let MSG_ERROR_DEVICE_NULL   = "Não foi possível efetuar uma conexão com o dispositivo pareado"
    let MSG_ERROR_GENERIC       = "Não foi possível efetuar a operação"
    
    // MARK: - UIViewController Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        PlugPag.sharedInstance().setVersionName("UnitTest", withVersion: "V001")
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - IBAction Buttons
    @IBAction func cancelTransaction(sender: UIButton) {
        
        sender.isEnabled = false
        UIUtils.showProgress()
        
        let ret = PlugPag.sharedInstance().initBTConnection()
        
        switch ret {
        case RET_OK:
            cancelTransaction()
            break
            
        case -2023:
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_DEVICE_NULL)")
            break
            
        default:
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_GENERIC)")
            break
        }
        
        UIUtils.hideProgress()
        sender.isEnabled = true
    }
    
    // MARK: - Business Methods
    func cancelTransaction() {
        
        let ret = PlugPag.sharedInstance().cancelTransaction()
        
        if (ret == RET_OK) {
            txtResult?.text = MSG_CANCEL_SUCESS
        }else{
            txtResult?.attributedText = UIUtils.formatTextError(error: Int(ret), msg: "\(MSG_ERROR_GENERIC)")
        }
    }

}
